s64da_benchmark_toolkit/correctness.py

```diff
@@ -2,24 +2,76 @@
 import logging
 import os
 
+from enum import Enum
+
 import numpy as np
 import pandas as pd
 
 from natsort import index_natsorted, order_by_index
-from pandas.io.formats.style import Styler
 
 LOG = logging.getLogger()
 
+
+class CorrectnessResult:
+    def __init__(self, status, detail=None, truth=[], result=[]):
+        self.status = status
+        self.detail = detail
+        self.truth = truth
+        self.result = result
+
+    @classmethod
+    def make_mismatch_result(cls, detail, truth, result):
+        return cls('MISMATCH', detail=detail, truth=truth, result=result)
+
+    @classmethod
+    def make_ok_result(cls):
+        return cls('OK')
+
+    @property
+    def is_ok(self):
+        return self.status == 'OK'
+
+    @property
+    def is_mismatch(self):
+        return self.status == 'MISMATCH'
+
+    @classmethod
+    def _tuple_to_html(cls, t):
+        html = ''
+        for row in t:
+            html += '<div>' + ' '.join([str(x) for x in row]) + '</div>'
+        return html
+
+    def __repr__(self):
+        return self.status
+
+    @classmethod
+    def format_html(cls, value):
+        if value.is_ok:
+            return value.status
+
+        # HTML here, since it'll be used for reporting to HTML
+        truth_html = value.truth.to_html() # CorrectnessResult._tuple_to_html(value.truth)
+        result_html = value.result.to_html() #CorrectnessResult._tuple_to_html(value.result)
+        return f'{value.status}<br /><div>{truth_html}</div><br /><div>{result_html}</div>'
```

Comment:
I'm against having html output as part of the logic. It would be better to separate the logic and visuals.

Reply:
I generally agree, but this here is the only exception ...

Reply:
ok for now. Should probably be revisited in the future, though.